presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -343,23 +350,23 @@ private boolean isTablesEnumeratingTable(SchemaTableName schemaTableName)
                             .filter(this::isLowerCase)
                             .map(table -> table.toLowerCase(ENGLISH))
                             .map(table -> new QualifiedObjectName(catalogName, prefix.getSchemaName().get(), table)))
-                    .filter(objectName -> metadata.getTableHandle(session, objectName).isPresent() || metadata.getView(session, objectName).isPresent())
+                    .filter(objectName -> !isColumnsEnumeratingTable(schemaTableName) || metadata.getTableHandle(session, objectName).isPresent() || metadata.getView(session, objectName).isPresent())
                     .filter(objectName -> !predicate.isPresent() || predicate.get().test(asFixedValues(objectName)))
                     .map(QualifiedObjectName::asQualifiedTablePrefix)
                     .collect(toImmutableSet());
         }
 
-        if (predicate.isPresent()) {
-            return prefixes.stream()
-                    .flatMap(prefix -> Stream.concat(
-                            metadata.listTables(session, prefix).stream(),
-                            metadata.listViews(session, prefix).stream()))
-                    .filter(objectName -> predicate.get().test(asFixedValues(objectName)))
-                    .map(QualifiedObjectName::asQualifiedTablePrefix)
-                    .collect(toImmutableSet());
+        if (!predicate.isPresent() || !isColumnsEnumeratingTable(schemaTableName)) {
```

Comment:
from cmt msg

> When table predicate is present one could enumerate all tables
that match the predicate. However, this is only beneficial ...

I understand this is beneficial for (eg) `columns`  and not `tables` table, but why not apply it universally? 
The fewer `if`s, the better.

Reply:
> I understand this is beneficial for (eg) columns and not tables table, but why not apply it universally?

Because this would lead to excessive metadata querying. First we would enumerate tables (prefixes) here. Then we would additionally query for table existence in `io.prestosql.metadata.MetadataManager#listTables`

We could simply do listing once when building `tables` table